submodules/telemetry/__init__.py

```diff
@@ -1,133 +1,145 @@
+        squishedpackets = ""
+        retVal = False
+
+        with self.packet_lock:
+            while len(self.log_stack) + len(self.err_stack) > 0:    # while there's stuff to pop off
+                next_packet = (str(self.err_stack[-1]) if len(self.err_stack) > 0 else str(self.log_stack[-1]))   # for the purposes of determining packet length
+                while len(base64.b64encode((squishedpackets + next_packet).encode('ascii'))) < self.config["telemetry"]["max_packet_size"] and len(self.log_stack) + len(self.err_stack) > 0:
+                    if len(self.err_stack) > 0: # prefer error messages over log messages
+                        squishedpackets += str(self.err_stack.pop())
+                    else:
+                        squishedpackets += str(self.log_stack.pop())
+                squishedpackets = base64.b64encode(squishedpackets.encode('ascii'))
+                # print(squishedpackets)
+                self.modules["aprs"].send(squishedpackets) # , radio) #FIXME currently just using APRS, what about radio_output?
+                retVal = True
+                squishedpackets = ""
+
+        return retVal
+
+    def clear_buffers(self) -> None:
+        """
+        Clear the telemetry buffers - clearing general_queue, the log, and error stacks.
+        :return: None
+        """
+        with self.packet_lock:
+            self.general_queue.clear()
+            self.log_stack.clear()
+            self.err_stack.clear()
+
+    def decide(self) -> None:
+        """
+        A thread method to constantly check general_queue for messages and process them if there are any.
+        :return: None
+        """
+        while True:
+            if len(self.general_queue) != 0:
+                with self.packet_lock:
+                    message = self.general_queue.popleft()
+                    if type(message) is str and message[0] == ';':
+                        self.modules["command_ingest"].enqueue(message)
+                        # print("Running command_ingest.enqueue(" + message + ")")
+                    elif type(message) is error.Error:
+                        self.err_stack.append(message)
+                    elif type(message) is log.Log:
+                        self.log_stack.append(message)
+                    else:  # Shouldn't execute (enqueue() should catch it) but here just in case
+                        self.logger.error("Message prefix invalid.")
+            sleep(1)
+
+    def start(self) -> None:
+        """
+        Starts the telemetry send thread
+        Config and modules must be initialized beforehand (config is through constructor, modules is through set_modules)
+        :return None
+        """
+        if self.config is None or self.modules is None:
+            raise RuntimeError("Config variable or self.modules empty and not initialized")
+
+        threadDecide = ThreadHandler(target=partial(self.decide), name="telemetry-decide")  # start telemetry 'decide' thread
```

Comment:
all objects should be initialized in `__init__()`

Reply:
Fixed in 149ce11